module-2/backend/week_two.md

```diff
@@ -26,20 +34,50 @@ end
 
 Now how would you find the owner of the team with an id of 4?
 
+Owner.where(team_id: 4)
+
 5. In a database that's holding students and teachers, what will be the relationship between students and teachers? Draw the schema diagram.
+Could be one to many or many to many depending on the type of school and grade level. 
+
+One to many: teacher will have many students, students belong to a teacher. Students have a teacher id along with the other attributes they have without the associations.
+
 6. Define foreign key, primary key, and schema.
+Foreign key of teacher id lives on student and the primary key would be that.
+
 7. Describe the relationship between a foreign key on one table and a primary key on another table.
+Foreign key lives on the table that belongs to the table that has the primary key.
+
 8. What are the parts of an HTTP response?
+A request is sent to the server which then references the corresponding controller action, which then reaches into the model and interaction with ORM, which returns the info to the controller so then the appropriate view (string) response can be sent to the client.
 
 
 ### Optional Questions
 
 1. Name your five favorite ActiveRecord methods (i.e. methods your models inherit from ActiveRecord) and describe what they do.
+.count = returns the count of all records for a specific table
+maximum = highest value in a column
+minimum = lowest value in a column
+calculation methods (average, sum)
+
 2. Name your three favorite ActiveRecord rake tasks and describe what they do.
+
 3. What two columns does `t.timestamps null: false` create in our database?
+created at and updated at
+
 4. In a database that's holding schools and teachers, what will be the relationship between schools and teachers?
+one to many or many to many
+
 5. In the same database, what will you need to do to create this relationship (draw a schema diagram)?
+rails g model Student name references:teacher
```

Comment:
This is how you would create a model/migrations, and more from the command line. What does this actually add to the Student table? What is the one piece of the students table that creates the relationship between students and teachers?